### DISCUSSION TITLE
Regarding using pygwalker on gitpod

### DISCUSSION THREAD

**Discussion by Danpr1984** [Kanaries/pygwalker#224] [Q&A]
Hi, I have been trying to use pygwalker on my data analysis project with pandas but it's not working. 

df = pd.read_csv('usdbudget.csv')
walker = pyg.walk(df, use_kernel_calc=True)

Should this be enough to see the tool popping up? I have tried many different things on a .ipynb file. Could I work with xlsx files as well? 

Such a cool tool but I have wasted a day trying to make it work and I am sure it's something silly. Hopefully someone could help.

**Comment by longxiaofei** [comment]
Hey, Can you provide me with your environment?
* python version
* jupyter version
* pygwalker version
* browser version

and after invoke `walker = pyg.walk(df, use_kernel_calc=True)`, what's response?

**Reply by longxiaofei** [reply]
`!python -m pip list | grep jupyter` will output version of jupyter tools.

**Reply by Danpr1984** [reply]
Package                   Version
------------------------- ---------
anyio                     3.7.1
argon2-cffi               21.3.0
argon2-cffi-bindings      21.2.0
arrow                     1.2.3
astor                     0.8.1
astroid                   2.15.6
asttokens                 2.2.1
async-lru                 2.0.4
attrs                     23.1.0
autopep8                  2.0.2
Babel                     2.12.1
backcall                  0.2.0
bandit                    1.7.5
beautifulsoup4            4.12.2
bleach                    6.0.0
certifi                   2023.7.22
cffi                      1.15.1
charset-normalizer        3.2.0
cmdstanpy                 1.1.0
comm                      0.1.4
contourpy                 1.1.0
convertdate               2.4.0
cryptography              41.0.3
cycler                    0.11.0
dacite                    1.8.1
debugpy                   1.6.7
decorator                 5.1.1
defusedxml                0.7.1
dill                      0.3.7
distlib                   0.3.7
docutils                  0.20.1
duckdb                    0.8.1
ephem                     4.1.4
et-xmlfile                1.1.0
executing                 1.2.0
fastjsonschema            2.18.0
feature-engine            1.6.1
filelock                  3.12.2
flake8                    6.1.0
fonttools                 4.42.1
fqdn                      1.5.1
gitdb                     4.0.10
GitPython                 3.1.32
holidays                  0.31
htmlmin                   0.1.12
idna                      3.4
ImageHash                 4.3.1
importlib-metadata        6.8.0
importlib-resources       6.0.1
install                   1.3.5
ipykernel                 6.25.1
ipython                   8.14.0
ipywidgets                8.1.0
isoduration               20.11.0
isort                     5.12.0
jaraco.classes            3.3.0
jedi                      0.19.0
jeepney                   0.8.0
Jinja2                    3.1.2
joblib                    1.3.2
json5                     0.9.14
jsonpointer               2.4
jsonschema                4.19.0
jsonschema-specifications 2023.7.1
jupyter_client            8.3.0
jupyter_core              5.3.1
jupyter-events            0.7.0
jupyter-lsp               2.2.0
jupyter_server            2.7.0
jupyter_server_terminals  0.4.4
jupyterlab                4.0.4
jupyterlab-pygments       0.2.2
jupyterlab_server         2.24.0
jupyterlab-widgets        3.0.8
keyring                   24.2.0
kiwisolver                1.4.4
lazy-object-proxy         1.9.0
littleutils               0.2.2
LunarCalendar             0.0.9
markdown-it-py            3.0.0
MarkupSafe                2.1.3
matplotlib                3.7.2
matplotlib-inline         0.1.6
mccabe                    0.7.0
mdurl                     0.1.2
mistune                   3.0.1
more-itertools            10.1.0
multimethod               1.9.1
mypy                      1.4.1
mypy-extensions           1.0.0
nbclient                  0.8.0
nbconvert                 7.7.3
nbformat                  5.9.2
nest-asyncio              1.5.7
networkx                  3.1
notebook                  7.0.2
notebook_shim             0.2.3
numpy                     1.23.5
openpyxl                  3.1.2
outdated                  0.2.2
overrides                 7.4.0
packaging                 23.1
pandas                    1.5.3
pandas-flavor             0.6.0
pandocfilters             1.5.0
parso                     0.8.3
patsy                     0.5.3
pbr                       5.11.1
pep8                      1.7.1
pexpect                   4.8.0
phik                      0.12.3
pickleshare               0.7.5
Pillow                    10.0.0
pingouin                  0.5.3
pip                       23.2.1
pipenv                    2023.7.23
pkginfo                   1.9.6
platformdirs              3.10.0
ppscore                   1.3.0
prometheus-client         0.17.1
prompt-toolkit            3.0.39
prophet                   1.1.4
psutil                    5.9.5
ptyprocess                0.7.0
pure-eval                 0.2.2
pyarrow                   13.0.0
pycodestyle               2.11.0
pycparser                 2.21
pydantic                  1.10.12
pydocstyle                6.3.0
pyflakes                  3.1.0
Pygments                  2.16.1
pygwalker                 0.3.5
pylama                    8.4.1
pylint                    2.17.5
PyMeeus                   0.5.12
pyparsing                 3.0.9
python-dateutil           2.8.2
python-json-logger        2.0.7
pytoolconfig              1.2.5
pytz                      2023.3
PyWavelets                1.4.1
PyYAML                    6.0.1
pyzmq                     25.1.0
readme-renderer           40.0
referencing               0.30.2
requests                  2.31.0
requests-toolbelt         1.0.0
rfc3339-validator         0.1.4
rfc3986                   2.0.0
rfc3986-validator         0.1.1
rich                      13.5.2
rope                      1.9.0
rpds-py                   0.9.2
scikit-learn              1.3.0
scipy                     1.11.2
seaborn                   0.12.2
SecretStorage             3.3.3
Send2Trash                1.8.2
setuptools                68.0.0
six                       1.16.0
smmap                     5.0.0
sniffio                   1.3.0
snowballstemmer           2.2.0
soupsieve                 2.4.1
sqlglot                   18.5.0
stack-data                0.6.2
statsmodels               0.14.0
stevedore                 5.1.0
tabulate                  0.9.0
tangled-up-in-unicode     0.2.0
terminado                 0.17.1
threadpoolctl             3.2.0
tinycss2                  1.2.1
tomlkit                   0.12.1
tornado                   6.3.2
tqdm                      4.66.1
traitlets                 5.9.0
twine                     4.0.2
typeguard                 2.13.3
typing_extensions         4.7.1
tzdata                    2023.3
uri-template              1.3.0
urllib3                   2.0.4
virtualenv                20.24.2
virtualenv-clone          0.5.7
visions                   0.7.5
wcwidth                   0.2.6
webcolors                 1.13
webencodings              0.5.1
websocket-client          1.6.1
wheel                     0.41.1
widgetsnbextension        4.0.8
wordcloud                 1.9.2
wrapt                     1.15.0
xarray                    2023.8.0
xgboost                   1.7.6
ydata-profiling           4.5.1
yellowbrick               1.5
zipp                      3.16.2

**Reply by Danpr1984** [reply]
Version 116.0.5845.187 (Official Build) (x86_64) chrome

**Reply by Danpr1984** [reply]
and after invoke walker = pyg.walk(df, use_kernel_calc=True), what's response?

This has been the response or it just keeps running but nothing happens. 
<pygwalker.api.pygwalker.PygWalker at 0x7fdf78d8ed90>

Thanks for the help!

**Comment by longxiaofei** [comment]
I will try to reproduce it.

**Comment by longxiaofei** [comment]
@Danpr1984 , I have the same environment as you, but I haven't reproduced it.

1. run code in your notebook, what's output? `Hello World`?
```
import ipywidgets
from IPython.display import display

display(ipywidgets.HTML("Hello World"))
```

2. run code, See if there is rendering UI?

```
# `env = "Jupyter"` is a version without communication
pyg.walk(df, env="Jupyter")
```

**Reply by Danpr1984** [reply]
![Screenshot 2023-09-15 at 09 08 25](https://github.com/Kanaries/pygwalker/assets/90693070/39de867d-2fba-4623-a084-7ebdc0be4a50)
![Screenshot 2023-09-15 at 09 09 45](https://github.com/Kanaries/pygwalker/assets/90693070/ab97b157-796c-47fa-bda9-b94e583966e5)

**Reply by Danpr1984** [reply]
It runs but it does not load anything. That's the strange part. If then I just run walker it keeps running and running....

**Reply by longxiaofei** [reply]
I guess `jupyter` version that you using is not compatible with `ipywidgets`.

First, try to update jupyterlab.

`pip install --upgrade jupyterlab`

If it still doesn't work. try to downgrade jupyter.

`pip install "jupyterlab<4.0.0"`

Don’t forget to restart jupyter after updating the version.

**Comment by Danpr1984** [comment]
![Screenshot 2023-09-15 at 10 04 13](https://github.com/Kanaries/pygwalker/assets/90693070/e266072a-f731-4cb7-8cc4-4b3663032082)

I have been currently working directly on gitpod, creating .ipynb files that allowed me to work on jupyter notebooks. Do you think this might be the problem? Usually everything else is working correctly.

**Reply by longxiaofei** [reply]
I ignore gitpod environment.

I guess there may be a problemin gitpod, Due to some communication protocols of jupyter require cloud vendors to adapt.

 I will try to reproduce it in gitpod.

**Reply by Danpr1984** [reply]
I fixed it. Thanks. Now I have a docker file and actually running on a jupyter notebook. The problem I had was because I was working directly in the editor which I believe is more convenient. I wonder if there could be a way to actually load pygwalker directly in the editor though......
![Screenshot 2023-09-15 at 10 16 25](https://github.com/Kanaries/pygwalker/assets/90693070/8c8f931a-06ea-4126-8d66-dd7a85132a6a)

**Reply by longxiaofei** [reply]
Normally, pygwalker can be run in vscode(local vscode).

**Reply by longxiaofei** [reply]
But I haven't tried it in vscode of cloud.

**Reply by Danpr1984** [reply]
Thanks! Did you manage to make it work in vscode?
